src/Aspire.Hosting/build/Aspire.Hosting.targets

```diff
@@ -12,18 +12,22 @@
     <ProjectCapability Include="Aspire" Condition=" '$(IsAspireHost)' == 'true' " />
   </ItemGroup>
 
+  <Target Name="_CreateServiceAppProjects">
+
+    <ItemGroup>
+      <_ServiceAppProject Include="@(ProjectReference->WithMetadataValue('_IsServiceApp', 'true'))" />
+    </ItemGroup>
+
+  </Target>
+  
   <!-- Generate the data structures for doing the codegen for service references -->
   <Target Name="CreateServiceMetadataSources"
-          DependsOnTargets="FindReferenceAssembliesForReferences"
-          Inputs="@(ReferencePathWithRefAssemblies)"
-          Outputs="@(ReferencePathWithRefAssemblies)">
+          DependsOnTargets="_CreateServiceAppProjects">
```

Comment:
NIT: Not really added in your changes, but I noticed this target is not private (doesn't start with "_") but the rest are, is there a special reasoning behind that?

Reply:
The reasoning is to support other languages (i.e. F#). See the comment here:

https://github.com/dotnet/aspire/blob/e344c7c1556bf7f52163ab6fe3ac94ef0c395835/src/Aspire.Hosting/build/Aspire.Hosting.targets#L108

New languages could plug-in to `WriteServiceMetadataSourcesDependsOn` (which I made "public" in this change), and then be `DependsOn="CreateServiceMetadataSources"`, and then generate the code for the language using the `ServiceMetadataSource` item.